evm_loader/program/src/entrypoint.rs

```diff
@@ -285,19 +286,26 @@ fn process_instruction<'a>(
                 account_storage.get_caller_account().ok_or(ProgramError::InvalidArgument)?,
                 &H160::from_slice(from_addr), trx.nonce, &trx.chain_id)?;
 
-            payment::check_collateral_account(
-                program_id,
-                // WARNING Only for tests when base is random
-                operator_sol_info,
-                collateral_pool_sol_info,
-                collateral_pool_index as usize)?;
             payment::transfer_from_operator_to_collateral_pool(
+                program_id,
+                collateral_pool_index,
                 operator_sol_info,
                 collateral_pool_sol_info,
                 system_info)?;
 
+            let fee = U256::from(1000_u64) * trx.gas_price * U256::from(1_000_000_000_u64);
+            transfer_token(
+                token_transfer_accounts,
+                user_eth_info,
+                operator_eth_info,
+                caller_info,
+                account_storage.get_caller_account().ok_or(ProgramError::InvalidArgument)?,
+                &fee)?;
+
             let trx_gas_limit = u64::try_from(trx.gas_limit).map_err(|_| ProgramError::InvalidInstructionData)?;
-            do_call(program_id, &mut account_storage, accounts, trx.call_data, trx.value, trx_gas_limit)
+            let _used_gas = do_call(program_id, &mut account_storage, accounts, trx.call_data, trx.value, trx_gas_limit)?;
```

Comment:
If I call `transfer_token` after `do_call` when delete account in Ethereum transaction called, I get "balance mismatch error", but if I move `transfer_token` before `do_call` - error disappears.
To avoid this error, deleted accounts could be marked in `do_call` and returned, and moving funds from them could be done after `transfer_token`. Or we could only mark deleted accounts as inactive just by writing `AccountData::Empty` to them.